### ISSUE TITLE
Raise exception instead of sys.exit()

### ISSUE THREAD

**Issue by b3nav** [edsu/xkcd2347#2]
Using the `GitHub` class in a separate project to retrieve the dependencies via the `get_dependencies` method would stop the program on an error. While technically the `sys.exit()` raises an exception, it's worth perhaps raising some other appropriate exception.

https://github.com/edsu/xkcd2347/blob/d6208ebd8d54367b983bf334541fd00f5ec7f088/xkcd2347.py#L110

**Comment by edsu** [comment]
Good idea! If you send a PR for what would be useful to you I will merge it.
